core/embed/boardloader/main.c

```diff
@@ -300,7 +322,9 @@ int main(void) {
          "invalid bootloader hash");
 
 #ifdef STM32U5
-  check_bootloader_version(hdr->monotonic);
+  uint8_t bld_version = get_bootloader_min_version();
+  ensure((bld_version >= hdr->monotonic) * sectrue, "BOOTLOADER DOWNGRADED");
```

Comment:
is this check backwards?

`bld_version` is the version from storage, `hdr->version` is the current one.
the current version must be _higher_ than the stored one, otherwise it's a downgrade, right?

Reply:
right. fixed https://github.com/trezor/trezor-firmware/commit/b4a3ad09e9a2c1306683c2a07606eea491875c6d